pyproject.toml

```diff
@@ -179,13 +179,14 @@ build-backend = "setuptools.build_meta"
 # a dev-only set of *tested* import/export target dataframe implementations
 # that can be converted to and from via narwhals
 dataframe = [
+    "numpy>=1.25.0",  # keep in sync with project.dependencies until gh-18782 is resolved
     # keep in sync with project.optional-dependencies.recommended
     # and project.optional-dependencies.all
     # and project.optional-dependencies.typing
     "narwhals>=1.42.0",
     "pyarrow>=14.0.2",
     "pandas>=2.0.2",
-    "polars>=1.0.0",
+    "polars>=1.18.0",
```

Comment:
In an early iteration I found that because our tests assume the existence of `polars.Int128`, this is the oldest version we actually support.